runtime/vm/jfr.cpp

```diff
@@ -965,6 +969,29 @@ jfrThreadCPULoadCallback(J9VMThread *currentThread, IDATA handlerKey, void *user
 	jfrThreadCPULoad(currentThread, currentThread);
 }
 
+void
+jfrClassLoadingStatistics(J9VMThread *currentThread)
+{
+	J9JavaVM *vm = currentThread->javaVM;
+	J9JFRClassLoadingStatistics *jfrEvent = (J9JFRClassLoadingStatistics *)reserveBuffer(currentThread, sizeof(J9JFRClassLoadingStatistics));
+
+	if (NULL != jfrEvent) {
+		initializeEventFields(currentThread, (J9JFREvent *)jfrEvent, J9JFR_EVENT_TYPE_CLASS_LOADING_STATISTICS);
+
+		UDATA unloadedClassCount = 0;
+		vm->memoryManagerFunctions->j9gc_get_cumulative_class_unloading_stats(currentThread, NULL, &unloadedClassCount, NULL);
+		jfrEvent->unloadedClassCount = (I_64)unloadedClassCount;
+
+		pool_state clState = {0};
```

Comment:
you should use the `allClassLoadersStartDo` API instead which acquires the correct class mutex

Reply:
`allClassLoadersNextDo` and `allClassLoadersEndDo`

Reply:
also, you need to release VM access before starting the classloading walk so you are not holding on to VM access while waiting for a lock

Reply:
fixed